### PR TITLE
models: add comments - feedback welcome

### PR DESCRIPTION
**Pull request by risicle**
I've found these notes useful. What do people think? Are they correct? Anyone have any additions? Amendments? Anywhere where I've completely got the wrong end of the stick?

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -422,6 +433,9 @@ def validate_role(self, key, value):
 
     @property
     def locked(self):
+        """
+            Whether account has had too many recent login attempts
```
Comment (TheDoubleK) [inline app/models.py]: `too many recent failed login attempts`

File: app/models.py [hunk 2]
```diff
@@ -399,7 +404,13 @@ class User(db.Model):
     password_changed_at = db.Column(db.DateTime, index=False, unique=False,
                                     nullable=False)
     logged_in_at = db.Column(db.DateTime, nullable=True)
+
+    # used to determine whether account is `locked`. field is reset upon successful login or can
+    # be reset manually to "unlock" an account.
     failed_login_count = db.Column(db.Integer, nullable=False, default=0)
+
+    # used by frontends to determine whether view access should be allowed, also
+    # used to police the validity of some operations by the api internally.
```
Comment (TheDoubleK) [inline app/models.py]: Interesting - I didn't really think the API cared about this internally. Can you remember where it's used (to save me doing the hard work to check it?)

File: app/models.py [hunk 3]
```diff
@@ -1167,4 +1189,5 @@ def filter_null_value_fields(obj):
 
 
 def generate_new_service_id(framework_slug):
+    # FIXME framework_slug parameter ignored??
```
Comment (TheDoubleK) [inline app/models.py]: Yes we should fix this.  In the past there were different generation rules for different frameworks.

File: app/models.py (hunk 2 above)
Comment (risicle) [inline app/models.py]: Actually "the validity of some operations" is misleading - it's places where e.g. an extra field is returned where the user is a supplier and so on mostly.

Comment (risicle) [inline app/models.py]: _Actually_, forget that - the place where it does that is in the `User` object serialization itself, which makes perfect sense. Can probably get away with removal of this sencence.

**Comment by risicle** [comment]
Ok let's see if this todo works:

- [ ] remove "police validity" comment
- [ ] change "failed login attempts" language

File: app/models.py [hunk 4]
```diff
@@ -422,6 +432,9 @@ def validate_role(self, key, value):
 
     @property
     def locked(self):
+        """
+            Whether account has had too many recent failed login attempts
+        """
```
Comment (pcraig3) [inline app/models.py]: Super nickpicky, but we don't care if they're recent.  

```Whether account has had too many failed login attempts``` is more accurate.

**Comment by pcraig3** [comment]
Yeah, generally seems okay to me.

File: app/models.py (hunk 4 above)
Comment (risicle) [inline app/models.py]: orly? ok.

**Comment by TheDoubleK** [comment]
Yep, let's get this merged :+1:

**Pull request merged**